pkg/kubernetes/node/v1alpha1/node.go

```diff
@@ -0,0 +1,119 @@
+package v1alpha1
+
+import "k8s.io/api/core/v1"
+
+// node holds the api's node object
+type node struct {
+	object *v1.Node
+}
+
+// node list holds the list of node instances
+type nodeList struct {
+	items []*node
+}
+
+// predicate defines an abstraction
+// to determine conditional checks
+// against the provided node instance
+type predicate func(*node) bool
+
+// predicateList holds the list of predicates
+type predicateList []predicate
+
+// listBuilder enables building an instance of nodeList
+type listBuilder struct {
+	list    *nodeList
+	filters predicateList
```

Comment:
using a map of predicate will help us to get more information on which predicate is responsible for filtering.
ref - https://github.com/openebs/maya/pull/1025
please have a look if it is useful for this usecase.

Reply:
@imazik IMO since most of the time, we will check whether the node is in the ready state or not. So it is not required for node package.